e2e/cypress/integration/account_settings/general/username_spec.js

```diff
@@ -0,0 +1,203 @@
+    it('MM-T2052 Username already taken', () => {
+        // # Click the General tab
+        cy.get('#generalButton').should('be.visible').click();
+
+        // # Edit the username field
+        cy.get('#usernameEdit').click();
+
+        // # Add the username to textfield contents
+        cy.get('#username').clear().type(`${otherUser.username}`);
+        cy.get('#saveSetting').click();
+
+        // * Check if element is present and contains expected text values
+        cy.get('#serverError').should('be.visible').should('contain', 'Unable to find the existing account to update.');
+
+        // # Click "x" button to close Account Settings modal
+        cy.get('#accountSettingsHeader > .close').click();
+    });
+
+    it('MM-T2053 Username w/ dot, dash, underscore still searches', () => {
+        let tempUser;
+
+        // # Create a temporary user
+        cy.apiCreateUser({prefix: 'temp'}).then(({user: user1}) => {
+            tempUser = user1;
+
+            cy.apiAddUserToTeam(testTeam.id, tempUser.id).then(() => {
+                cy.apiAddUserToChannel(testChannel.id, tempUser.id);
+            });
+
+            // # Login the temporary user
+            cy.apiLogin(tempUser);
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+            cy.toAccountSettingsModal();
+
+            // # Step 1
+            // # Click the General tab
+            cy.get('#generalButton').should('be.visible').click();
+
+            // # Edit the username field
+            cy.get('#usernameEdit').click();
+
+            // # Step 2
+            // # Add the username to textfield contents
+            const newTempUserName = 'a-' + tempUser.username;
+            cy.get('#username').clear().type(newTempUserName);
+            cy.get('#saveSetting').click();
+
+            // * Check that we return to the Account Settings panel
+            cy.get('#generalButton').should('be.visible');
+
+            // # Click "x" button to close Account Settings modal
+            cy.get('#accountSettingsHeader > .close').click();
+
+            // # Step 3
+            // * Verify that we've logged in as the temp user
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+            cy.get('#headerUsername').should('contain', '@' + newTempUserName);
+
+            // # Step 4
+            // # Clear then type @
+            cy.get('#post_textbox').should('be.visible').clear().type('@');
+
+            // * Verify that the suggestion list is visible
+            cy.get('#suggestionList').should('be.visible');
+
+            // # Type user name
+            cy.get('#post_textbox').type(newTempUserName);
+            cy.get('#post_textbox').type('{enter}{enter}');
+
+            // # Check that the user name has been posted
+            cy.getLastPostId().then((postId) => {
+                cy.get(`#postMessageText_${postId}`).should('contain', newTempUserName);
+            });
```

Comment:
I think it's fine to at-mention self but still missing the last step:
- "4. Click "@" beside the search box to view recent mentions" then verify "Recent mentions button searches for updated username"